src/views/DatasetManager/components/CreateDatasetButton/CreateSubDatasetButtonHook.js

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+import { useCallback, useMemo } from 'react';
+import { useCreateRunner } from '../../../../state/hooks/RunnerHooks';
+import { useSubDataSourceRunTemplates, useSolutionData } from '../../../../state/hooks/SolutionHooks';
+import { ArrayDictUtils, SolutionsUtils } from '../../../../utils';
+
+export const useSubDatasetCreationParameters = () => {
+  const createRunner = useCreateRunner();
+  const solutionData = useSolutionData();
+  const customSubDataSourceRunTemplates = useSubDataSourceRunTemplates();
+
+  const dataSourceRunTemplates = useMemo(() => {
+    const parameters = solutionData.parameters;
+    const runTemplatesParameters = solutionData.runTemplatesParametersIdsDict;
+    const dataSourcesWithParameters = customSubDataSourceRunTemplates.map((dataSource) => ({
+      ...dataSource,
+      parameters: parameters.filter((parameter) => runTemplatesParameters[dataSource.id].includes(parameter.id)),
+    }));
+
+    const runTemplates = {};
```

Comment:
Is it necessary to set `runtemplate` as _object null_ instead of _null_ ?

Reply:
Yes, because the next line tries to create keys in this object, and we can't do that if it's `null`:
`   [...]    runTemplates[runTemplate.id] = runTemplate));`